Tools/glib/dependencies/apt

```diff
@@ -2,14 +2,25 @@
 
 # If the package $1 is available, prints it. Otherwise prints $2.
 # Useful for handling when a package is renamed on new versions of Debian/Ubuntu.
-function aptIfElse {
+aptIfElse() {
     if apt-cache show $1 &>/dev/null; then
         echo $1
     else
         echo $2
     fi
 }
 
+aptIfExists() {
+    local ret=$(apt show "$1" 2>/dev/null)
```

Comment:
@dpino I'm not going to pretend to understand quite why this is the case, but per [this comment](https://stackoverflow.com/questions/2556103/is-it-possible-to-get-the-exit-code-from-a-subshell#comment107482312_2556122) this formulation needs to look instead like `local ret; ret=$(apt show "$1" 2>/dev/null)` (but maybe with a newline? see below), otherwise `$?` always ends up 0 and running `Tools/gtk/install-dependencies` will try to install a bunch of nonexistent packages like "libenchant-dev" and this breaks the install request such that important packages like "ruby" won't be installed.

I ran into this when trying to [update the searchfox wubkat tree to use the new explicit support for Ubuntu 22.04 by using the install-dependencies script](https://github.com/mozsearch/mozsearch-mozilla/blob/30cc471be0f2e78628df955994b40c0f39b3f899/wubkat/setup#L70) and the configure step failed because there was no ruby because the call to "install-dependencies" only installed a few python packages.  I added `set -x` to the script so I could see what the commands were and it was clear from the output that the evaluation of `$?` in the next line was incorrectly 0 so I searched and found the above stackoverflow comment and when I changed the script to instead have:
```
    local ret;
    ret=$(apt show "$1" 2>/dev/null)
```
the script then installed everything as expected.  I only tried this formulation with the newline, I didn't try on the same line.

Note that I think it's also the case that apt is explicitly not supposed to be used for scripting so it might be better to use apt-cache show here like is used above?  Like when used in a sub-shell it complains "WARNING: apt does not have a stable CLI interface. Use with caution in scripts."

Reply:
Thanks for the pointer. This was indeed a source of errors that was preventing running `install-dependencies` in different platforms.

The reason why the code isn't working as it's writen is because in Bash an assigment always returns exit code 0. So, as a rule of thumb, is generally wrong in Bash to initialize a variable and inmediately after check the result of its assigned expression. That's why it's necessary to break down the statement into declaration of var, assigned expression and evaluation of result. I remember @aperezdc actually [warned me about this once](https://bugs.webkit.org/show_bug.cgi?id=228276#c5) in another review of this code.

Unrelated to this but perhaps you already noticed you also needed to manually install the package `unidef`. It's pending to be added in apt dependencies file.

Reply:
Shell-wise: Aha, that does make sense, thanks!  That will probably make a good addition to https://github.com/mozsearch/mozsearch/blob/master/docs/bash-scripting-cheatsheet.md

Build-wise: Indeed, the build broke because of (I'm assuming there's a missing "f" typo in your current comment) `unifdef` not being installed and then I (very gratefully!) realized that support for Ubuntu 22.04 had been added so I thought I'd try using it.  I've restored the "wubkat" indexer behavior to fail if the build step fails, so once the build starts working again wubkat should go back to always having semantic data available at the trade-off of the webkit tree seeming to have fatal build errors surprisingly frequently so data will be stale when that happens.

Reply:
Right, the name of the package is `unifdef`. I noticed it has been already added in `Tools/gtk/dependencies/atk` in the patch that introduced this dependency [253528@main](https://github.com/webkit/webkit/compare/f76a7e30e574%5E..f76a7e30e574). So nothing pending to be done here.

We're currently in the middle of an upgrade of webkit-search.igalia.com to Ubuntu 22.04 (plus upgrade of the mozsearch checkout). I've been reviewing your [wubkat scripts](https://github.com/mozsearch/mozsearch-mozilla/tree/master/wubkat). They're very useful, I learned many things. There are other things, on the hand, that could be improved or you'd like to reconsinder. I think I will open a PR to comment on the code.

Reply:
A PR would be fantastic, thank you!  I very much am not familiar with building webkit and mainly was just trying to get things working at all, and clearly leaned very heavily on your existing (public) config and the config in emilio's branch.

The canonical searchfox discussion channel is https://chat.mozilla.org/#/room/#searchfox:mozilla.org (and maybe it's federated?) if you want to chat more about searchfox outside of specific pull requests, etc.  I'll actually be working this week on finishing implementing a TOML configuration file that will allow for custom mappings for what's a test file and allow for more normalized ingestion of JSON files with per-file info, plus support for custom templating to augment the rendered HTML as an evolution of the current test info / bugzilla mapping / coverage data ingestion mechanism we use for mozilla-central.

Reply:
Created bug and opened PR to fix the aptIfExists error and update system library dependencies https://github.com/webkit/webkit/pull/3525